### PR TITLE
Add CMake compatibility

### PR DESCRIPTION
**Pull request by MartinDelille**
I'm thinking about moving from qmake to cmake and I'll wondered if you would accept this small addition to make it compatible ?

### CODE REVIEW

**Comment by gamecreature** [comment]
Hi @MartinDelille, Thank you for you contribution.
I'm comparing this with the previous pull request which has been waiting for ages.
This on doesn't change the headers, so that a plus :-)
I see it depends on Qt5, is possible to change it so it allows other versions?

Could you explain how to use it with CMake in readme.md?

**Comment by Jihadist** [comment]
@MartinDelille ping :)

**Comment by MartinDelille** [comment]
Sorry I didn't saw the first notification! I will try to update it to support Qt5 and Qt6. Would it be fine for you ? I don't know if Qt4 is needed.

**Comment by gamecreature** [comment]
@MartinDelille, no problem. I didn't test Qt4 with latest release, so you can leave that one out. (Qt4 support is in the other branches). Thanks for your time!

**Comment by MartinDelille** [comment]
I fixed qt6 compatibility. Unfortunately it doesn't work on Qt6 due to some deprecated function (but maybe it is due to the fact that I am to strong with my warning error).

File: QtAwesome/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+project(QtAwesome)
+
+add_library(${PROJECT_NAME}
```
Comment (Jihadist) [inline QtAwesome/CMakeLists.txt:3]: Looks like library cannot build as shared on Windows. What about to add CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS or throw fatal error for such case or force static lib as qmake project?

**Review by Jihadist — APPROVED** [review]

File: QtAwesome/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
+project(QtAwesome)
+
+add_library(QtAwesome
```
Comment (Jihadist) [inline QtAwesome/CMakeLists.txt:3]: Looks like you forgot the root cmakelists with find_package or you forgot find_package here

File: QtAwesome/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+target_include_directories(QtAwesome
+  INTERFACE ${PROJECT_SOURCE_DIR}
+)
+
+target_link_libraries(QtAwesome PUBLIC
+  Qt${QT_VERSION_MAJOR}::Widgets
+)
+
+include(GenerateExportHeader)
```
Comment (Jihadist) [inline QtAwesome/CMakeLists.txt:17]: I checked this. But it's not a full replace for export_all. It just generates a header with directives. You need to mark exported classes, or you will have something same:
`QtAwesome.obj : error LNK2001: unresolved external symbol "public: virtual struct QMetaObject const * __cdecl fa::QtAwesome::metaObject(void)const " (?metaObject@QtAwesome@fa@@UEBAPEBUQMetaObject@@XZ)
QtAwesome.obj : error LNK2001: unresolved external symbol "public: virtual void * __cdecl fa::QtAwesome::qt_metacast(char const *)" (?qt_metacast@QtAwesome@fa@@UEAAPEAXPEBD@Z)
QtAwesome.obj : error LNK2001: unresolved external symbol "public: virtual int __cdecl fa::QtAwesome::qt_metacall(enum QMetaObject::Call,int,void * *)" (?qt_metacall@QtAwesome@fa@@UEAAHW4Call@QMetaObject@@HPEAPEAX@Z)
QtAwesome.obj : error LNK2019: unresolved external symbol "int __cdecl qInitResources_QtAwesomeFree(void)" (?qInitResources_QtAwesomeFree@@YAHXZ) referenced in function "void __cdecl qtawesome_init_resources(void)" (?qtawesome_init_resources@@YAXXZ)
QtAwesome.obj : error LNK2001: unresolved external symbol "public: static struct QMetaObject const fa::QtAwesomeAnimation::staticMetaObject" (?staticMetaObject@QtAwesomeAnimation@fa@@2UQMetaObject@@B)
QtAwesomeAnim.obj : error LNK2001: unresolved external symbol "public: virtual struct QMetaObject const * __cdecl fa::QtAwesomeAnimation::metaObject(void)const " (?metaObject@QtAwesomeAnimation@fa@@UEBAPEBUQMetaObject@@XZ)
QtAwesomeAnim.obj : error LNK2001: unresolved external symbol "public: virtual void * __cdecl fa::QtAwesomeAnimation::qt_metacast(char const *)" (?qt_metacast@QtAwesomeAnimation@fa@@UEAAPEAXPEBD@Z)
QtAwesomeAnim.obj : error LNK2001: unresolved external symbol "public: virtual int __cdecl fa::QtAwesomeAnimation::qt_metacall(enum QMetaObject::Call,int,void * *)" (?qt_metacall@QtAwesomeAnimation@fa@@UEAAHW4Call@QMetaObject@@HPEAPEAX@Z)`

File: QtAwesome/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+project(QtAwesome)
```
Comment (Jihadist) [inline QtAwesome/CMakeLists.txt:1]: No cmake_minimum_required command is present

File: QtAwesomeSample/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+project(QtAwesomeSample)
+
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core)
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Widgets)
+
+set(CMAKE_AUTOUIC ON)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+
+add_subdirectory(../QtAwesome QtAwesome)
```
Comment (Jihadist) [inline QtAwesomeSample/CMakeLists.txt:11]: Ah, okey I see. What about to build QtAwesome without QtAwesomeSample?

Comment (MartinDelille) [reply QtAwesomeSample/CMakeLists.txt:11]: Usually, if I need QtAwesome, I have a Qt program already doing the find_package work.

Comment (Jihadist) [reply QtAwesomeSample/CMakeLists.txt:11]: Sometimes you need to build library as a main project. Maybe ci/cd. Maybe you wanna add some tests

File: QtAwesome/CMakeLists.txt (hunk 3 above)
Comment (MartinDelille) [reply QtAwesome/CMakeLists.txt]: Reverted here: 8b3cf90

File: QtAwesome/CMakeLists.txt (hunk 4 above)
Comment (MartinDelille) [reply QtAwesome/CMakeLists.txt:1]: Fixed here: 54c87a4

File: QtAwesomeSample/CMakeLists.txt (hunk 5 above)
Comment (MartinDelille) [reply QtAwesomeSample/CMakeLists.txt:11]: Fixed here: ab15dd6

**Comment by MartinDelille** [comment]
I added an unrelated fix here: a0ad43a

Let me know if you prefer a separate PR for it!

**Review by Jihadist — APPROVED** [review]

**Comment by Jihadist** [comment]
It works well with Qt5 and Qt6, thanx :)

**Review by gamecreature — APPROVED** [review]

**Comment by gamecreature** [comment]
@Jihadist, @MartinDelille, thanks for contributing!

**Pull request merged**